### PR TITLE
[Docs] Add a sample to clarify animation callback

### PR DESCRIPTION
**Pull request by jsdario**
## Motivation

Even having worked long with React Native, I am pretty new to animations, so it took me a while to figure out how to trigger some code after an animation had finished. After some investigation, it
was evident I did not ready as in depth as I am supposed to, but this can be a problem for many
who, like me, try to search quick through docs and other resources.

Here more cases:
https://stackoverflow.com/questions/38053071/react-native-animated-complete-event
https://github.com/facebook/react-native/issues/3212

## Test Plan
Read the proposed sample, and see if it is adequate.

## Release Notes
 [DOCS] [ENHANCEMENT] [AnimatedImplementation.js] - Adds a sample for animation callbacks

### CODE REVIEW

**Comment by hramos** [comment]
Can you rebase? We just landed a commit that shifts documentation to plain old markdown files in the `docs/` directory. These jsdoc comments will be stripped out soon and I'd hate to lose your change.

**Comment by jsdario** [comment]
Sure thing @hramos! Just did so :)

File: docs/animated.md [hunk 1]
```diff
@@ -66,6 +66,16 @@ invoked with `{finished: true}`. If the animation is done because `stop()`
 was called on it before it could finish (e.g. because it was interrupted by a
 gesture or another animation), then it will receive `{finished: false}`.
 
+```javascript
+this.animateValue.spring({}).start(function onComplete({finished}) {
```
Comment (janicduplessis) [inline docs/animated.md]: Use arrow function here

File: docs/animated.md [hunk 2]
```diff
@@ -66,6 +66,16 @@ invoked with `{finished: true}`. If the animation is done because `stop()`
...
+  if (finished === true) {
```
Comment (janicduplessis) [inline docs/animated.md]: if (finished) {

File: docs/animated.md [hunk 3]
```diff
@@ -66,6 +66,16 @@ invoked with `{finished: true}`. If the animation is done because `stop()`
...
+    console.log('Animation was stopped')
```
Comment (janicduplessis) [inline docs/animated.md]: "Animation was completed" maybe?

**Comment by jsdario** [comment]
Hi @janicduplessis, thanks for the comments. Just updated based on feedback 😄 I can squash commits if necessary.

**Comment by janicduplessis** [comment]
@jsagorin Thanks, you don't have to squash commits it will be done automatically when your PR lands.

**Comment by janicduplessis** [comment]
@facebook-github-bot shipit

**Comment by janicduplessis** [comment]
@hramos Can you check why this failed to import?

**Comment by jsdario** [comment]
Can I do something to help with this? @hramos @janicduplessis

**Comment by hramos** [comment]
Trying to land again. I couldn't find any particular reason this failed to import.

**Pull request closed without merging**